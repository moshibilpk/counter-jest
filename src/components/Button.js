
import { TouchableHighlight, Text, StyleSheet } from 'react-native'
import React from 'react'

type PropsType = {|
    onPress?: () => Promise <void> | void,
    title: string | Element <typeof Text>,
  |};

const Button = ({ buttonName, onPress }:PropsType): Element<typeof TouchableHighlight>  => {
    return (
        <TouchableHighlight style={styles.counterButton} onPress={onPress}>
            <Text style={styles.labelStyle}>{buttonName}</Text>
        </TouchableHighlight>
    )
}

export default Button

const styles = StyleSheet.create({
    counterButton: {
        backgroundColor: '#000',
        padding: 10,
        borderRadius: 5
    },
    labelStyle: {
        color: '#fff'
    }
})

