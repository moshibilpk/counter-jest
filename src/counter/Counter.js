// @flow
import {View, TouchableHighlight, Text, StyleSheet} from 'react-native';
import React from 'react';
import type {Element} from 'react';
import Button from '../components/Button';
import type {Node} from 'react';

type PropsType = {|
  incrementCount?: () => Promise<void> | void,
  decrementCount?: () => Promise<void> | void,
  count: number,
|};

const Counter = ({count, incrementCount, decrementCount}: PropsType): Node => {
  return (
    <View style={styles.container}>
      <Button buttonName={'Increment'} onPress={incrementCount} />
      <Text style={styles.counter}>{count}</Text>
      {count !== 0 ? (
        <Button buttonName={'Decrement'} onPress={decrementCount} />
      ) : null}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  counter: {
    fontSize: 20,
    color: '#000',
    margin: 5,
  },
});

export default Counter;
