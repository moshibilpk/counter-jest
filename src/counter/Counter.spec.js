// @flow
import React from 'react';
import {findAllWithType, findWithType} from 'react-shallow-testutils';
import ShallowRenderer from 'react-test-renderer/shallow';
import Counter from './Counter';
import Button from '../components/Button';

const renderCounter = () => {
  const renderer = new ShallowRenderer();
  renderer.render(
    <Counter
      count={count}
      incrementCount={incrementCount}
      decrementCount={decrementCount}
    />,
  );
  return renderer.getRenderOutput();
};

let incrementCount: () => void,
  decrementCount: () => void,
  count: number,
  incrementButton = 'Increment',
  decrementButton = 'Decrement';

describe('counter', () => {
  beforeEach(() => {
    incrementCount = jest.fn();
    decrementCount = jest.fn();
    count = 0;
  });
  test('render correctly', () => {
    const result = renderCounter();
    expect(result).toMatchSnapshot();
  });
  test('0 visible and buttonNames are correct', () => {
    const result = renderCounter();
    expect(result.props.children[0].props.buttonName).toEqual(incrementButton);
    expect(result.props.children[1].props.children).toBe(0);
    expect(result.props.children[2].props.buttonName).toEqual(decrementButton);
  });
});
