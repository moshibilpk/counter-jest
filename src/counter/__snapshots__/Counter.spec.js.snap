// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Button
    buttonName="Increment"
    onPress={[MockFunction]}
  />
  <Text
    style={
      Object {
        "color": "#000",
        "fontSize": 20,
        "margin": 5,
      }
    }
  >
    0
  </Text>
  <Button
    buttonName="Decrement"
    onPress={[MockFunction]}
  />
</View>
`;
